File: Golang/SingleNumberII/SingleNumberII_test.go
package SingleNumberII

import "testing"

func TestSingleNumber(t *testing.T) {

	if singleNumber([]int{1,}) == 1 {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if singleNumber([]int{1, 1, 1,}) == 0 {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if singleNumber([]int{1, 1, 1, 2,}) == 2 {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if singleNumber([]int{1, 1, 1, 2, 2, 2, 3,}) == 3 {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

}

File: Python/SingleNumberIITest.py
from unittest import TestCase

from SingleNumberII import SingleNumberII


class TestSingleNumberII(TestCase):
    def test_singleNumber(self):
        s = SingleNumberII()

        self.assertTrue(s.singleNumber([1]) == 1)

        self.assertTrue(s.singleNumber([1, 1, 1]) == 0)

        self.assertTrue(s.singleNumber([1, 1, 1, 2]) == 2)

        self.assertTrue(s.singleNumber([1, 1, 1, 2, 2, 2, 3]) == 3)
